plugins/statuslines/lualine.nix

```diff
@@ -1,177 +1,278 @@
+    in
+    {
+      options = {
+        icons_enabled = defaultNullOpts.mkBool true ''
+          Whether to enable icons for all components.
 
-      componentSeparators = mkSeparatorsOption {
-        leftDefault = "";
-        rightDefault = "";
-      };
+          This option is also available on individual components to control whether they should display icons.
+        '';
 
-      sectionSeparators = mkSeparatorsOption {
-        leftDefault = "";
-        rightDefault = "";
-      };
+        theme = defaultNullOpts.mkNullable (
+          with types; either str attrs
+        ) "auto" "The theme to use for lualine-nvim.";
 
-      disabledFiletypes = {
-        statusline = helpers.defaultNullOpts.mkListOf types.str [ ] ''
-          Only ignores the ft for statusline.
-        '';
+        component_separators = mkSeparatorsOption {
+          left = "";
+          right = "";
+        };
 
-        winbar = helpers.defaultNullOpts.mkListOf types.str [ ] ''
-          Only ignores the ft for winbar.
-        '';
-      };
+        section_separators = mkSeparatorsOption {
+          left = "";
+          right = "";
+        };
 
-      ignoreFocus = helpers.defaultNullOpts.mkListOf types.str [ ] ''
-        If current filetype is in this list it'll always be drawn as inactive statusline and the
-        last window will be drawn as active statusline.
+        disabled_filetypes = defaultNullOpts.mkNullableWithRaw' {
+          description = ''
+            Filetypes in which to disable lualine.
+            Allows you to specify filetypes that you want to only disable on specific components.
+          '';
+          pluginDefault = { };
+          type =
+            with types;
+            either (listOf str) (submodule {
+              freeformType = attrsOf anything;
+
+              options = {
+                statusline = defaultNullOpts.mkListOf str [ ] ''
+                  Hide the statusline component on specified filetypes.
+                '';
+
+                winbar = defaultNullOpts.mkListOf str [ ] ''
+                  Hide the winbar component on specified filetypes.
+                '';
+              };
+            });
+        };
 
-        For example if you don't want statusline of your file tree / sidebar window to have active
-        statusline you can add their filetypes here.
-      '';
+        ignore_focus = defaultNullOpts.mkNullableWithRaw' {
+          type = types.listOf types.str;
+          pluginDefault = [ ];
+          description = ''
+            A list of filetypes that should always show an "unfocused" statusline.
+
+            If the focused window's filetype is in this list, then the most
+            recently focused window will be drawn as the active statusline.
+          '';
+          example = ''
+            [
+              "neo-tree"
+              "nvim-tree"
+              "mini-files"
+            ]
+          '';
```

Comment:
```suggestion
          example =[
            "neo-tree"
            "nvim-tree"
            "mini-files"
          ];
```
:wink:

Reply:
Lol sorry, was thinking it had to be a string for some reason.

Reply:
It can be, if you use `literalExpression` or `literalMD`